src/lib/mdns/tests/TestTxtFields.cpp

```diff
@@ -372,35 +381,146 @@ void TestFillNodeDataFromTxt(nlTestSuite * inSuite, void * inContext)
     // Pairing instruction
     sprintf(key, "PI");
     sprintf(val, "hint");
-    FillNodeDataFromTxt(GetSpan(key), GetSpan(val), &filled);
+    FillNodeDataFromTxt(GetSpan(key), GetSpan(val), filled);
     NL_TEST_ASSERT(inSuite, strcmp(filled.pairingInstruction, "hint") == 0);
     memset(filled.pairingInstruction, 0, sizeof(filled.pairingInstruction));
     NL_TEST_ASSERT(inSuite, NodeDataIsEmpty(filled));
 
     // Pairing hint
     sprintf(key, "PH");
     sprintf(val, "1");
-    FillNodeDataFromTxt(GetSpan(key), GetSpan(val), &filled);
+    FillNodeDataFromTxt(GetSpan(key), GetSpan(val), filled);
     NL_TEST_ASSERT(inSuite, filled.pairingHint == 1);
     filled.pairingHint = 0;
     NL_TEST_ASSERT(inSuite, NodeDataIsEmpty(filled));
 }
 
+bool NodeDataIsEmpty(const ResolvedNodeData & nodeData)
+{
+    return nodeData.mPeerId == PeerId{} && nodeData.mAddress == Inet::IPAddress::Any && nodeData.mPort == 0 &&
+        nodeData.mMrpRetryIntervalIdle == 0 && nodeData.mMrpRetryIntervalActive == 0 && !nodeData.mSupportsTcp;
+}
+
+// Test CRI
+void TxtFieldMrpRetryIntervalIdle(nlTestSuite * inSuite, void * inContext)
+{
+    char key[4];
+    char val[8];
+    ResolvedNodeData nodeData;
+
+    // Minimum
+    sprintf(key, "CRI");
+    sprintf(val, "1");
+    FillNodeDataFromTxt(GetSpan(key), GetSpan(val), nodeData);
+    NL_TEST_ASSERT(inSuite, nodeData.mMrpRetryIntervalIdle == 1);
+
+    // Maximum
+    sprintf(key, "CRI");
+    sprintf(val, "3600000");
+    FillNodeDataFromTxt(GetSpan(key), GetSpan(val), nodeData);
+    NL_TEST_ASSERT(inSuite, nodeData.mMrpRetryIntervalIdle == 3600000);
+
+    // Test no other fields were populated
+    nodeData.mMrpRetryIntervalIdle = 0;
+    NL_TEST_ASSERT(inSuite, NodeDataIsEmpty(nodeData));
+
+    // Invalid CRI => fallback to 0
+    sprintf(key, "CRI");
+    sprintf(val, "-1");
+    FillNodeDataFromTxt(GetSpan(key), GetSpan(val), nodeData);
+    NL_TEST_ASSERT(inSuite, nodeData.mMrpRetryIntervalIdle == 0);
+
+    // Invalid CRI => fallback to 0
+    sprintf(key, "CRI");
+    sprintf(val, "asdf");
+    FillNodeDataFromTxt(GetSpan(key), GetSpan(val), nodeData);
+    NL_TEST_ASSERT(inSuite, nodeData.mMrpRetryIntervalIdle == 0);
+}
```

Comment:
Please test at least the following the following additional CRI values: "123abc", "4000000", "1234567890", "9876543210", "0x20", "0700".

Reply:
Added more checks.